main-cron: failure on end-to-end iceberg sink v2 test (release)

### Describe the bug

https://buildkite.com/risingwavelabs/main-cron/builds/1998#018e1846-fa8a-4950-b617-60da80bf8f0f

### Error message/log

```text
Command line is [sqllogictest -p 4566 -d dev test_case/iceberg_sink_upsert.slt]
Executing sql: SELECT * FROM demo_db.demo_table ORDER BY id, v1 ASC
Row(id=1, v1=1, v2=50, v3='1-50', v4=datetime.date(2022, 3, 11))
Row(id=1, v1=3, v2=2, v3='3-2', v4=datetime.date(2022, 3, 13))
Row(id=1, v1=8, v2=2, v3='8-2', v4=datetime.date(2022, 3, 18))
Row(id=1, v1=13, v2=2, v3='13-2', v4=datetime.date(2022, 3, 13))
    tc.assertEqual(len(df), len(rows))
  File "/usr/lib/python3.10/unittest/case.py", line 845, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python3.10/unittest/case.py", line 838, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 4 != 7
```


### To Reproduce

_No response_

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

--- a/e2e_test/iceberg/test_case/iceberg_sink_upsert.slt
+++ b/e2e_test/iceberg/test_case/iceberg_sink_upsert.slt
@@ -27,16 +27,11 @@ CREATE SINK s6 AS select mv6.id as id, mv6.v1 as v1, mv6.v2 as v2, mv6.v3 as v3,
 statement ok
 INSERT INTO t6 VALUES (1, 1, 2, '1-2', '2022-03-11'), (1, 2, 2, '2-2', '2022-03-12'), (1, 3, 2, '3-2', '2022-03-13'), (1, 5, 2, '5-2', '2022-03-15'), (1, 8, 2, '8-2', '2022-03-18'), (1, 13, 2, '13-2', '2022-03-13'), (1, 21, 2, '21-2', '2022-03-21');
 
-statement ok
-FLUSH;
-
-sleep 5s
+sleep 10s
 
 statement ok
 INSERT INTO t6 VALUES (1, 1, 50, '1-50', '2022-03-11');
 
-statement ok
-FLUSH;
 
 sleep 10s
 
@@ -44,9 +39,6 @@ query I
 select count(*) from t6;
 ----
 7
-
-statement ok
-FLUSH;
 
 sleep 10s
 

